### PR TITLE
feat: stop storing algo category

### PR DESCRIPTION
**Pull request by AlexandrePicosson**
## Description

The category field will should not be consumed anymore, we will keep accepting it but we will not do anything with it and expose a default value to consumers.

Blocked by: 
- https://github.com/Substra/substra-backend/pull/460
- https://github.com/Substra/substra/pull/285

## How has this been tested?

End to end tests

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [x] documentation was updated

### CODE REVIEW

**Comment by AurelienGasser** [comment]
Is it intentional not to include a DB migration?

**Comment by AlexandrePicosson** [comment]
It's a work in progress, the migration is coming. But I'm not sure I should delete the existing categories I'll probably just stop storing new ones.

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by AlexandrePicosson** [comment]
/e2e --refs substra-backend=feat/algo_unknown,substra=feat/no-category

**Comment by AlexandrePicosson** [comment]
The CI do not pass, but because of another mistake unrelated, we can consider it green 🟢

**Comment by AlexandrePicosson** [comment]
e2e tests results:
```
============ 110 passed, 7 skipped, 7 warnings in 571.28s (0:09:31) ============
```

**Review by thibowk — APPROVED** [review]
LGTM 👏!

Maybe add `/* intentionally empty */` to the down migration as it has previously been done.

File: lib/service/performance.go [hunk 1]
```diff
@@ -66,15 +66,6 @@ func (s *PerformanceService) RegisterPerformance(newPerf *asset.NewPerformance,
 		return nil, errors.NewBadRequest(fmt.Sprintf("cannot register performance for task with status %q", task.Status.String()))
 	}
 
-	algo, err := s.GetAlgoService().GetAlgo(newPerf.MetricKey)
```
Comment (mblottiere) [inline lib/service/performance.go:69]: We still need to check that the referenced algo does exist

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+**WARNING**: Some migrations in this version are destructive once applied you will not be able to restore algo categories.
+
+### Changed
+- Algo.category: You should not rely on categories anymore, all algo categories will be returned as UNKNOWN
```
Comment (mblottiere) [inline CHANGELOG.md:11]:
```suggestion
- (BREAKING) Algo.category: do not rely on categories anymore, all algo categories will be returned as UNKNOWN
```

File: lib/service/performance.go (hunk 1 above)
Comment (AlexandrePicosson) [reply lib/service/performance.go:69]: Good catch.

**Pull request merged**